File: 프로그래머스/lv1/12943. 콜라츠 추측/콜라츠 추측.kt
class Solution {
    fun solution(num: Int): Int {
        var num = num.toLong()
        var result = 0

        while(num != 1.toLong()) {
            if((num % 2).toInt() == 0) {
                num /= 2
            }
            else {
                num = num*3 + 1
            }
            result++

            if(result > 500) {
                result = -1
                break
            }
        }
        return result
    }
}

File: 프로그래머스/1/12943. 콜라츠 추측/콜라츠 추측.swift
func solution(_ num:Int) -> Int {
    if num == 1 { return 0 }
    
    var count = 0
    var tempNum = num
    var arr: [Int] = []
    
    while count < 501 {
        count += 1
        
        if tempNum % 2 == 0 {
            tempNum /= 2
        } else {
            tempNum = tempNum * 3 + 1
        }
        
        if tempNum == 1 {
            return count
        }
        
        if arr.contains(tempNum) {
            return -1
        }
    }
    
    return -1
}